src/util-conf.c

```diff
@@ -63,3 +64,30 @@ TmEcode ConfigCheckLogDirectory(char *log_dir)
     }
     SCReturnInt(TM_ECODE_OK);
 }
+
+/**
+ * \brief Find the configuration node for a specific device.
+
+ * Basically hunts through the list of maps for the first one with a
+ * key of "interface", and a value of the provided interface.
+ *
+ * \param node The node to start looking for the device
+ *     configuration. Typically this would be something like the af-packet
+ *     or pf-ring node.
+ *
+ * \param iface The name of the interface to find the config for.
+ */
+ConfNode *ConfFindDeviceConfig(ConfNode *node, const char *iface)
+{
+    ConfNode *if_node, *item;
+    const char key[] = "interface";
+    TAILQ_FOREACH(if_node, &node->head, next) {
+        TAILQ_FOREACH(item, &if_node->head, next) {
+            if (!(strcmp(item->name, key) && strcmp(item->val, iface))) {
```

Comment:
Ah, yes. This is flawed. It works with the expected YAML, but there is YAML that could make it fail.